contributing/development/cpp_usage_guidelines.rst

```diff
@@ -53,18 +53,69 @@ as Godot provides its own data types (among other things).
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``Packed*Array`` 📜    | ``std::vector``          | Equivalent to ``Vector``, e.g. ``PackedColorArray = Vector<Color>``.                |
+|                        |                          | Only a limited list of packed array types are available                             |
+|                        |                          | (use ``TypedArray`` otherwise).                                                     |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``Span``               | ``std::span``            | Represents read-only access to a contiguous array without needing to copy any data. |
+|                        |                          | See `pull request description <https://github.com/godotengine/godot/pull/100293>`__ |
+|                        |                          | for details.                                                                        |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``HashSet``            | ``std::set``             | **Use this as the "default" set type.**                                             |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``RBSet``              | ``std::set``             | Uses a `red-black tree <https://en.wikipedia.org/wiki/Red-black_tree>`__            |
+|                        |                          | for faster access.                                                                  |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``HashMap``            | ``std::map``             | **Use this as the "default" map type.** Iteration is different from ``std::map``    |
+|                        |                          | (``HashMap`` is insertion-ordered, while ``std::map`` is key-ordered).              |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``AHashMap``           | ``std::map``             | Array-based implementation of a hash map.                                           |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``OAHashMap``          | ``std::unordered_map``   | Does not preserve insertion order. Generally faster than ``HashMap`` and            |
+|                        |                          | ``AHashMap``.                                                                       |
```

Comment:
I'm not sure it's faster than `AHashMap`. I don't quite remember how they compare in terms of speed.

Reply:
I remember reduz mentioning this a few years ago (along with many pull requests that replaced uses of `HashMap` and `AHashMap` with `OAHashMap` for performance reasons. It's possible the performance situation has changed since then, so I've removed the note.